bam_readcount_helper.py

```diff
@@ -96,14 +101,14 @@ def filter_sites_in_hash(region_list, bam_file, ref_fasta, sample, output_dir, i
 
 if len(rc_for_snp.keys()) > 0:
     region_file = generate_region_list(rc_for_snp)
-    filter_sites_in_hash(region_file, bam_file, ref_fasta, sample, output_dir, False, min_mapping_qual, min_base_qual)
+    filter_sites_in_hash(region_file, bam_file, ref_fasta, sample, output_dir, False, prefix, min_mapping_qual, min_base_qual)
```

Comment:
Instead of changing the parameters to `filter_sites_in_hash` I propose that you create a variable `prefixed_sample` and pass that along instead of the `sample` name. To make that work, on line 46-49 you would do:
```
if prefix == 'NOPREFIX':
    prefixed_sample = sample
else:
    prefixed_sample = '_'.join([prefix, sample])
```
You can then also use `prefixed_sample` to set your `output_file` below.

Reply:
This makes sense as better coding. I make the change as requested.